### PR TITLE
python3Packages.cairocffi: 1.5.1 -> 1.6.1

### PR DESCRIPTION
**Pull request by jleightcap**
## Description of changes

https://github.com/Kozea/cairocffi/releases/tag/v1.6.1

Upstream moved to `pyproject.toml`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/cairocffi/default.nix [hunk 1]
```diff
@@ -60,19 +53,19 @@ buildPythonPackage rec {
     fontDirectories = [ freefont_ttf ];
   };
 
-  propagatedNativeBuildInputs = [ cffi ];
+  propagatedNativeBuildInputs = [ cffi flit-core ];
```
Comment (tjni) [inline pkgs/development/python-modules/cairocffi/default.nix:56]: I can't wrap my head around what `propagatedNativeBuildInputs` is propagating, but I feel like the build backend a package chooses should not be propagated. Can that go in `nativeBuildInputs` instead?

File: pkgs/development/python-modules/cairocffi/default.nix [hunk 2]
```diff
@@ -22,13 +23,14 @@
 
 buildPythonPackage rec {
   pname = "cairocffi";
-  version = "1.4.0";
+  version = "1.6.1";
+  format = "pyproject";
 
   disabled = pythonOlder "3.5";
```
Comment (tjni) [inline pkgs/development/python-modules/cairocffi/default.nix:29]: This is now disabled for Python older than 3.7.

**Review by tjni — COMMENTED** [review]
Looks pretty good.

**Comment by tjni** [comment]
I just realized it will need to be merged with https://github.com/NixOS/nixpkgs/commit/b9012fdfc6772418017b797a63a3002a756e634e, which updates cairocffi to 1.5.1. This is currently in staging-next and probably a few days away from making it onto master.

**Comment by dotlambda** [comment]
Master now has 1.5.1. Please fix the conflicts.

File: pkgs/development/python-modules/cairocffi/default.nix [hunk 3]
```diff
@@ -45,11 +45,10 @@ buildPythonPackage rec {
   ];
 
   nativeBuildInputs = [
+    flit-core
     setuptools
```
Comment (tjni) [inline pkgs/development/python-modules/cairocffi/default.nix:49]: Can you check if `setuptools` can be removed (from here and from the argument list at the top of the file)?

**Review by tjni — COMMENTED** [review]
Given the number of rebuilds, can you try [the instructions here](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging) and rebase this change onto the `staging` branch?

In particular, I find this part hidden in those instructions important:

> As an additional precautionary measure, you should temporarily mark the PR as draft for the duration of the operation. This reduces the probability of mass-pinging people.

so even if it doesn't work the first time, you can play around with it to get it right. (Apologies for including this if you already know how to do this.)

**Pull request merged**